pkg/network/protocols/testutil/serverutils.go

```diff
@@ -54,3 +54,39 @@ func RunDockerServer(t *testing.T, serverName, dockerPath string, env []string,
 		}
 	}
 }
+
+func RunHostServer(t *testing.T, command []string, env []string, serverStartRegex *regexp.Regexp) {
+	if len(command) < 1 {
+		t.Fatalf("command not set %v host server", command)
+	}
+	t.Helper()
+
+	cmd := exec.Command(command[0], command[1:]...)
+	serverName := cmd.String()
+	patternScanner := NewScanner(serverStartRegex, make(chan struct{}, 1))
+
+	cmd.Stdout = patternScanner
+	cmd.Stderr = patternScanner
+	cmd.Env = append(cmd.Env, env...)
+	go func() {
+		require.NoErrorf(t, cmd.Run(), "could not start %s on host", serverName)
```

Comment:
generally, calling `t.Fatal` on another goroutine is discouraged
> A test ends when its Test function returns or calls any of the methods FailNow, Fatal, Fatalf, SkipNow, Skip, or Skipf. Those methods, as well as the Parallel method, must be called only from the goroutine running the Test function.

Reply:
You could maybe use a `chan error` here, and add it to the `select` below.

Reply:
The next PR will change this behavior removing the Fatal and returning a boolean

Reply:
the problem with calling from a go routine is that it can end with tests being hang forever and the entire CI will be stuck